pkg/clients/logging/logger.go

```diff
@@ -160,10 +163,19 @@ func (lc EdgeXLogger) log(logLevel string, msg string, args ...interface{}) {
 		args = append(args, "msg", msg)
 	}
 
+	levelLock.RLock()
 	if lc.levelLoggers[logLevel] == nil {
+		levelLock.RUnlock()
+		// Upgrade lock to write
+		levelLock.Lock()
 		lc.levelLoggers[logLevel] = log.WithPrefix(lc.rootLogger, "level", logLevel)
+		levelLock.Unlock()
```

Comment:
If we were to pull this in, I think the only locking you'd need is this writable lock since the levelLoggers map is being modified. All of the other read lock mgmt shouldn't be necessary.

But in order to prove that out I need to be able to reliably reproduce and as I said in my comment on issue #1025 I cannot.

Reply:
On my machine at least, I need both locks to avoid the race condition warnings.